managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -1174,43 +1174,49 @@ public void testSetProperties() throws Exception {
         properties.put("key1", "value1");
         properties.put("key2", "value2");
         properties.put("key3", "value3");
-        ledger.setProperties(properties);
+        ledger.setProperties(properties, true);
         assertEquals(ledger.getProperties(), properties);
 
         Map<String, String> newProperties = new HashMap<>();
         newProperties.put("key4", "value4");
         newProperties.put("key5", "value5");
         newProperties.put("key6", "value6");
-        ledger.setProperties(newProperties);
+        ledger.setProperties(newProperties, true);
+        assertEquals(ledger.getProperties(), newProperties);
+
+        ledger.setProperties(properties, false);
+        newProperties.putAll(properties);
```

Comment:
Please also try to add entries in the ledger, to verify that we're not fenced off from the metadata updates. 

Change the settings to force multiple rollovers, eg: 

```java
ManagedLedger ledger = factory.open("my_test_ledger",
     new ManagedLedgerConfig().setMaxEntriesPerLedger(1));
```

Reply:
OK.